src/_pytest/python.py

```diff
@@ -156,6 +156,15 @@ def pytest_configure(config):
 @hookimpl(trylast=True)
 def pytest_pyfunc_call(pyfuncitem):
     testfunction = pyfuncitem.obj
+    iscoroutinefunction = getattr(inspect, "iscoroutinefunction", None)
+    if iscoroutinefunction is not None and iscoroutinefunction(testfunction):
+        msg = "test function {} is a coroutine and not natively supported.\n"
```

Comment:
shouldn't this be a format table warning as constant in the warnigns module

Reply:
We don't do that for `PytestWarnings` in general.

IIRC the objective of concentrating `PytestDeprecationWarning`s in `deprecated.py` is to have all of those type of warnings in a single place as a reminder of what we need to remove when the time for a major release comes around.

We don't have the same need for `PytestWarning`'s, because we don't intend to ever remove them, so I don't think it is worth concentrating them in a single place.

Let me know what you think.

Reply:
We should address this in a separate PR if we want to move this forward, as it will affect other warnings as well. 👍